packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/hardforks.ts

```diff
@@ -177,42 +177,43 @@ describe("Eth module - hardfork dependant tests", function () {
           );
         });
       });
-    });
-
-    describe("Transaction type validation by hardfork", function () {
-      describe("Without EIP155 nor access list", function () {
-        useProviderAndCommon("tangerineWhistle");
 
-        it("Should reject an eth_sendRawTransaction if signed with EIP-155", async function () {
-          const spuriousDragonCommon = this.common.copy();
-          spuriousDragonCommon.setHardfork("spuriousDragon");
-
-          const signedTx = getSampleSignedTx(spuriousDragonCommon);
-          const serialized = bufferToRpcData(signedTx.serialize());
+      describe("In a hardfork with EIP-1559", function () {
+        useProviderAndCommon("london");
 
+        it("Should validate the chain id if sent to eth_sendTransaction using an eip-1559 fields", async function () {
+          const [sender] = await this.provider.send("eth_accounts");
           await assertInvalidArgumentsError(
             this.provider,
-            "eth_sendRawTransaction",
-            [serialized],
-            "Trying to send an EIP-155 transaction"
+            "eth_sendTransaction",
+            [
+              {
+                from: sender,
+                to: sender,
+                chainId: numberToRpcQuantity(1),
+                maxFeePerGas: numberToRpcQuantity(10e9),
+              },
+            ],
+            "Invalid chainId"
           );
         });
 
-        it("Should reject an eth_sendRawTransaction if the tx uses an access list", async function () {
-          const berlinCommon = this.common.copy();
-          berlinCommon.setHardfork("berlin");
-
-          const signedTx = getSampleSignedAccessListTx(berlinCommon);
+        it("Should validate the chain id in eth_sendRawTransaction using an eip-1559 tx", async function () {
+          const signedTx = getSampleSignedEIP1559Tx();
```

Comment:
I'm confused about how this works. It's because `getSampleSignedEIP1559Tx` is not receiving a `common` and therefore the chainId in the raw tx is invalid? I wonder if we can be more explicit here.

Reply:
Makes sense. I made the common required in every helper.